features/timeseries/src/main/java/org/opennms/netmgt/timeseries/samplewrite/TimeseriesPersisterFactory.java

```diff
@@ -44,25 +44,46 @@
 import org.opennms.netmgt.rrd.RrdRepository;
 
 import com.codahale.metrics.MetricRegistry;
+import org.opennms.netmgt.timeseries.TimeseriesStorageManager;
+import org.opennms.netmgt.timeseries.stats.StatisticsCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 
 /**
  * Factory for {@link TimeseriesPersister}.
  *
  * @author jwhite
  */
 public class TimeseriesPersisterFactory implements PersisterFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(TimeseriesPersisterFactory.class);
 
     private final TimeseriesWriter timeseriesWriter;
     private final MetaTagDataLoader metaTagDataLoader;
     private final Cache<ResourcePath, Set<Tag>> configuredAdditionalMetaTagCache;
     private final MetricRegistry registry;
 
     @Inject
-    public TimeseriesPersisterFactory(final TimeseriesWriter timeseriesWriter,
-                                      final MetaTagDataLoader metaTagDataLoader,
+    public TimeseriesPersisterFactory(final MetaTagDataLoader metaTagDataLoader,
+                                      final StatisticsCollector stats,
+                                      @Named("timeseriesStorageManager") final TimeseriesStorageManager timeseriesStorageManager,
                                       @Named("timeseriesPersisterMetaTagCache") final CacheConfig cacheConfig,
-                                      @Named("timeseriesMetricRegistry") MetricRegistry registry) {
-        this.timeseriesWriter = timeseriesWriter;
+                                      @Named("timeseriesMetricRegistry") MetricRegistry registry,
+                                      @Named("timeseries.ring_buffer_size") Integer ringBufferSize,
+                                      @Named("timeseries.writer_threads") Integer numWriterThreads,
+                                      @Named("timeseries.backend") String backend,
+                                      @Named("timeseries.offheap.batchSize") final int batchSize,
+                                      @Named("timeseries.offheap.path") final String path,
+                                      @Named("timeseries.offheap.maxFileSize") final Long maxFileSize) {
+        if (OffheapTimeSeriesWriter.OFFHEAP_NAME.equals(backend)) {
+            this.timeseriesWriter = new OffheapTimeSeriesWriter(timeseriesStorageManager, ringBufferSize,
+                    numWriterThreads, batchSize, path, maxFileSize, registry);
+        } else {
+            this.timeseriesWriter = new RingBufferTimeseriesWriter(timeseriesStorageManager, stats, ringBufferSize,
+                    numWriterThreads, registry);
+        }
+        LOG.debug("Writer: {}", this.timeseriesWriter);
```

Comment:
Not sure if this shouldn't be info? Its called only once and might be needed in the log, but not sure...